ckan/logic/action/create.py

```diff
@@ -1000,7 +1001,17 @@ def user_create(context: Context,
     upload.upload(uploader.get_max_image_size())
 
     if not context.get('defer_commit'):
-        model.repo.commit()
+        try:
+            model.repo.commit()
+        except exc.IntegrityError as e:
+            if e.orig.pgcode == _PG_ERR_CODE['unique_violation']:
+                session.rollback()
+                raise ValidationError(cast(ErrorDict, {
+                    'constraints': ['The email address \'{email}\' belongs to '
```

Comment:
I think it's better to add an error to the `email` key, not to the `constraints`. In the WebUI it will attach the error message to the email field and, after all, the error is raised because of unacceptable email, so it will be fair

Reply:
Yep, my bad